app/models/authenticator/httpd.rb

```diff
@@ -120,7 +120,7 @@ def user_details_from_headers(username, request)
                     :lastname  => request.headers['X-REMOTE-USER-LASTNAME'],
                     :email     => request.headers['X-REMOTE-USER-EMAIL'],
                     :domain    => request.headers['X-REMOTE-USER-DOMAIN']}
-      [user_attrs, (request.headers['X-REMOTE-USER-GROUPS'] || '').split(/[;:,]/)]
+      [user_attrs, (CGI.unescape(request.headers['X-REMOTE-USER-GROUPS']) || '').split(/[;:,]/)]
```

Comment:
we need to check that the request.headers['X-REMOTE-USER-GROUPS'] is present? first before we can do the CGI.unescape, otherwise, ugly stack trace.

Reply:
@abellotti Good catch. I had the check `|| ''` on the wrong side of the `)`
I just posted the commit withe the fix and an additional test in the spec to
confirm the handling of `nil` groups.

Thannk you! JoeV